compiler/src/language_server/hints/constant_evaluator.rs

```diff
@@ -75,46 +78,40 @@ impl ConstantEvaluator {
         let vm = &self.vms[module];
         let mut hints = vec![];
 
-        if let Status::Panicked { reason } = vm.status() {
+        if let vm::Status::Panicked { reason } = vm.status() {
             if let Some(hint) = panic_hint(db, module.clone(), vm, reason) {
                 hints.push(hint);
             }
         };
-        if module.to_possible_paths().is_some() {
-            module.dump_associated_debug_file("trace", &vm.tracer.format_call_tree(&vm.heap));
-        }
 
-        for entry in vm.tracer.log() {
-            let (id, value) = match entry {
-                TraceEntry::ValueEvaluated { id, value } => {
-                    if &id.module != module {
-                        continue;
-                    }
-                    let ast_id = match db.hir_to_ast_id(id.clone()) {
-                        Some(ast_id) => ast_id,
-                        None => continue,
-                    };
-                    let ast = match db.ast(module.clone()) {
-                        Some((ast, _)) => (*ast).clone(),
-                        None => continue,
-                    };
-                    let ast = match ast.find(&ast_id) {
-                        Some(ast) => ast,
-                        None => continue,
-                    };
-                    if !matches!(ast.kind, AstKind::Assignment(_)) {
-                        continue;
-                    }
-                    (id.clone(), value)
+        let TearDownResult { heap, tracer, .. } = vm.clone().tear_down();
+        for entry in &tracer.events {
+            if let EventData::ValueEvaluated { id, value } = &entry.data {
```

Comment:
You can remove one indentation level with `let … else …`:

```suggestion
            let EventData::ValueEvaluated { id, value } = &entry.data else { continue; };
```

Reply:
Nice, I didn't know about this syntax!

Reply:
It's pretty new, the feature gate just got removed. https://github.com/rust-lang/rust/issues/87335